engine/src/execution_graph/logic_controllers/BatchAggregationProcessing.h

```diff
@@ -25,30 +25,58 @@ class ComputeAggregateKernel : public kernel {
 	ComputeAggregateKernel(std::size_t kernel_id, const std::string & queryString, std::shared_ptr<Context> context, std::shared_ptr<ral::cache::graph> query_graph)
 		: kernel{kernel_id, queryString, context, kernel_type::ComputeAggregateKernel} {
         this->query_graph = query_graph;
+
+        std::tie(this->group_column_indices, this->aggregation_input_expressions, this->aggregation_types,
+            this->aggregation_column_assigned_aliases) = ral::operators::parseGroupByExpression(this->expression);
 	}
 
     bool can_you_throttle_my_input() {
 		return true;
 	}
 
 	virtual kstatus run() {
-		CodeTimer timer;
+		
+        CodeTimer timer;
         CodeTimer eventTimer(false);
-
-        std::vector<std::string> aggregation_input_expressions, aggregation_column_assigned_aliases;
-        std::tie(this->group_column_indices, aggregation_input_expressions, this->aggregation_types,
-            aggregation_column_assigned_aliases) = ral::operators::parseGroupByExpression(this->expression);
+    
+        eventTimer.start();
 
         bool ordered = false; // If we start using sort based aggregations this may need to change
-        BatchSequence input(this->input_cache(), this, ordered);
+        BatchSequenceBypass input(this->input_cache(), this); // WSM TODO, need to make BatchSequenceBypass take in "ordered" parameter
         int batch_count = 0;
         while (input.wait_for_next()) {
 
-            this->output_cache()->wait_if_cache_is_saturated();
-            auto batch = input.next();
+            // this->output_cache()->wait_if_cache_is_saturated();
+            std::vector<std::unique_ptr<ral::cache::CacheData>> batches;
+            batches.push_back(input.next());
+
+            kstatus status = run_batch(std::move(batches), batch_count);
+
+            batch_count++;
+        }
 
+        logger->debug("{query_id}|{step}|{substep}|{info}|{duration}|kernel_id|{kernel_id}||",
+                    "query_id"_a=context->getContextToken(),
+                    "step"_a=context->getQueryStep(),
+                    "substep"_a=context->getQuerySubstep(),
+                    "info"_a="ComputeAggregate Kernel Completed",
+                    "duration"_a=timer.elapsed_time(),
+                    "kernel_id"_a=this->get_id());
+
+        return kstatus::proceed;
+    }
+
+    virtual kstatus run_batch(std::vector<std::unique_ptr<ral::cache::CacheData>> batches, int batch_count) {
```

Comment:
why are you passing batch_count if you aren't using it?

Reply:
batch_count here is actually more like a place holder. The idea is that there will be more info passed along with the batches, with which you can setup priorities.